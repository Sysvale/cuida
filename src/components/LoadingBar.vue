<template>
	<div>
		<div class="loading-bar__container">
			<div class="loading-bar" />
		</div>
	</div>
</template>

<script>
export default {
	name: 'CdsLoadingBar',
};
</script>

<style lang="scss" scoped>
@use '../assets/sass/tokens/index' as tokens;

@mixin animation($animation) {
	-webkit-animation: $animation;
	-o-animation: $animation;
	animation: $animation;
}

@-webkit-keyframes loading-bar-stripes {
	from {
		background-position: 40px 0;
	}
	to {
		background-position: 0 0;
	}
}

@keyframes loading-bar-stripes {
	from {
		background-position: 40px 0;
	}
	to {
		background-position: 0 0;
	}
}

.loading-bar__container {
	display: flex;
	overflow: hidden;
	border-radius: tokens.$border-radius-small;
	height: 12px;
	transform: rotate(180deg);
}

.loading-bar {
	background-color: tokens.$gp-500;
	background-size: 40px 40px;
	width: 100%;
	@include animation(loading-bar-stripes 2s linear infinite);

	background-image: -webkit-linear-gradient(
		-45deg,
		#ffffff26 25%,
		transparent 25%,
		transparent 50%,
		#ffffff26 50%,
		#ffffff26 75%,
		transparent 75%,
		transparent
	);

	background-image: -o-linear-gradient(
		-45deg,
		#ffffff26 25%,
		transparent 25%,
		transparent 50%,
		#ffffff26 50%,
		#ffffff26 75%,
		transparent 75%,
		transparent
	);

	background-image: linear-gradient(
		-45deg,
		#ffffff26 25%,
		transparent 25%,
		transparent 50%,
		#ffffff26 50%,
		#ffffff26 75%,
		transparent 75%,
		transparent
	);
}
</style>
