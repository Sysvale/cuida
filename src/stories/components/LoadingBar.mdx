import { Meta, Story, Props, Controls, Canvas, Preview } from '@storybook/blocks';
import LoadingBar from '../../components/LoadingBar.vue';
import * as LoadingBarStories from './LoadingBar.stories';

<Meta of={LoadingBarStories} />

export const Template = (args) => ({
	components: { CdsLoadingBar: LoadingBar },
	setup() {
		return { args };
	},
	template: /*html*/ `
		<center>
			<cds-loading-bar
				v-bind="args"
			/>
		</center>
	`,
});

# LoadingBar

### LoadingBars é um componente utilizado para indicar que o progresso de uma operação ainda não foi finalizado.

***

<br />

## Quando usar:

* For necessário impedir a visualização e navegação do conteúdo de páginas enquanto houver alguma pendência de operação.

<br />

## Quando não usar:

* A visualização e navegação não puder ser interrompida.

<br />

## Preview

<Canvas>
  <Story of={LoadingBarStories.LoadingBar} />
</Canvas>

<Controls />
