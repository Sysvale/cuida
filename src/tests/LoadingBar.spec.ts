import { describe, test, expect } from 'vitest';
import LoadingBar from '../components/LoadingBar.vue';
import { mount } from '@vue/test-utils';

describe('LoadingBar', () => {
	test('renders correctly', async () => {
		const wrapper = mount(LoadingBar, {
			props: {},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
