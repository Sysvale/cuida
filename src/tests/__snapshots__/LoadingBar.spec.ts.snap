// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingBar > renders correctly 1`] = `
"<div data-v-cc77174b="">
  <div data-v-cc77174b="" class="loading-bar__container">
    <div data-v-cc77174b="" class="loading-bar"></div>
  </div>
</div>"
`;
